airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -115,51 +114,86 @@ public void start() {
+            databaseUser,
             databasePassword, databaseUrl, airbyteVersion),
         new PersistStateActivityImpl(workspaceRoot, configRepository));
+
+    final Worker replicationOrchestratorWorker =
+        factory.newWorker(TemporalJobType.REPLICATION_ORCHESTRATOR.name(), getWorkerOptions(maxWorkers.getMaxSyncWorkers()));
+    replicationOrchestratorWorker.registerWorkflowImplementationTypes(ReplicationOrchestratorWorkflowImpl.class);
+    replicationOrchestratorWorker.registerActivitiesImplementations(
+        new ReplicationActivityImpl(workerConfigs, orchestratorProcessFactory, secretsHydrator, workspaceRoot, workerEnvironment, logConfigs,
+            databaseUser,
+            databasePassword, databaseUrl, airbyteVersion));
+
     factory.start();
   }
 
-  private static ProcessFactory getProcessBuilderFactory(final Configs configs) throws IOException {
+  private static ProcessFactory getJobProcessFactory(final Configs configs) throws IOException {
+    final WorkerConfigs workerConfigs = new WorkerConfigs(configs);
+
     if (configs.getWorkerEnvironment() == Configs.WorkerEnvironment.KUBERNETES) {
       final ApiClient officialClient = Config.defaultClient();
       final KubernetesClient fabricClient = new DefaultKubernetesClient();
       final String localIp = InetAddress.getLocalHost().getHostAddress();
       final String kubeHeartbeatUrl = localIp + ":" + KUBE_HEARTBEAT_PORT;
       LOGGER.info("Using Kubernetes namespace: {}", configs.getKubeNamespace());
-      return new KubeProcessFactory(configs.getKubeNamespace(), officialClient, fabricClient, kubeHeartbeatUrl, configs.getTemporalWorkerPorts());
+      return new KubeProcessFactory(workerConfigs, configs.getKubeNamespace(), officialClient, fabricClient, kubeHeartbeatUrl);
     } else {
       return new DockerProcessFactory(
+          workerConfigs,
           configs.getWorkspaceRoot(),
           configs.getWorkspaceDockerMount(),
           configs.getLocalDockerMount(),
-          configs.getDockerNetwork());
+          configs.getDockerNetwork(),
+          false);
+    }
+  }
+
+  private static ProcessFactory getOrchestratorProcessFactory(final Configs configs) throws IOException {
+    if (configs.getWorkerEnvironment() == Configs.WorkerEnvironment.KUBERNETES) {
+      // there is no difference for kube
+      return getJobProcessFactory(configs);
+    } else {
+      return new DockerProcessFactory(
+          new WorkerConfigs(configs),
+          configs.getWorkspaceRoot(),
+          configs.getWorkspaceDockerMount(),
+          configs.getLocalDockerMount(),
+
+          // this needs to point at the Docker network Airbyte is running on, not the host network or job
+          // runner network
+          // otherwise it can't talk with the db/minio
+          "airbyte_default", // todo: should this be configurable via envconfigs?
```

Comment:
Would appreciate input on this. Should it be configurable with the default of "airbyte_default"? Is this a safe change to roll out?

Reply:
Was this previously exposed? I looked at the dockerfile and it doesn't seem to be the case.

If it isn't, I would skip for now and create an issue. We can implement this later if/when users ask for it.

Reply:
To be clear, master currently just uses the default network, right? So this is potentially a significant change.

I think the reason we were using the host network was because you need to for at least hitting anything you want to connect to on the host machine. e.g. being able to pull from a db that is running locally.

Reply:
This is only for spinning up orchestrator pods, NOT for spinning up connector pods.

The orchestrator needs to speak to at least minio, so it seems like the natural choice. I'll just leave as is without configuration options and add support if people are doing something weird like a host-network minio they want to connect to.